packages/app/src/systems/Account/components/BalanceWidget/BalanceWidget.tsx

```diff
@@ -0,0 +1,152 @@
+import { cssObj } from "@fuel-ui/css";
+import { Avatar, Box, Flex, Icon, IconButton, Text } from "@fuel-ui/react";
+import type { ReactNode } from "react";
+
+import type { Account } from "../../types";
+
+import { BalanceWidgetLoader } from "./BalanceWidgetLoader";
+
+import { shortAddress } from "~/systems/Core";
+
+type BalanceWidgetWrapperProps = {
+  children: ReactNode;
+};
+
+export function BalanceWidgetWrapper({ children }: BalanceWidgetWrapperProps) {
+  return (
+    <Box css={styles.balanceWidgetWrapper}>
+      <Box css={styles.backgroundShadow}>&nbsp;</Box>
+      <Box css={styles.backgroundFront}>&nbsp;</Box>
+      <Flex css={styles.contentWrapper}>{children}</Flex>
+    </Box>
+  );
+}
+
+export type BalanceWidgetProps = {
+  account: Account;
+  isHidden?: boolean;
+};
+
+export function BalanceWidget({ account, isHidden }: BalanceWidgetProps) {
+  async function handleCopy() {
+    await navigator.clipboard.writeText(account.address);
+  }
+
+  return (
+    <BalanceWidgetWrapper>
+      <Flex direction="column" align="center">
+        <Avatar
+          size="lg"
+          name={account.name}
+          /**
+           * TODO: Need to add Avatar.Generated here when it's done on @fuel-ui
```

Comment:
We already have the Avatar.Generated https://fuellabs.github.io/fuel-ui/?path=/story/ui-avatar--generated